### ISSUE TITLE
Failed to load the plugin 'installer_omod.dll' while start

### ISSUE THREAD

**Issue by GF-Huang** [ModOrganizer2/modorganizer#1413]
![image](https://user-images.githubusercontent.com/4510984/108551304-b8bac280-732a-11eb-9824-7550c6c16595.png)

![image](https://user-images.githubusercontent.com/4510984/108551345-c83a0b80-732a-11eb-8e9e-76d39d26a11f.png)

**Comment by Holt59** [comment]
You probably need to install .NET framework: https://dotnet.microsoft.com/download/dotnet-framework (version 4.8 should do it).

**Comment by GF-Huang** [comment]
Not this issue. 

![image](https://user-images.githubusercontent.com/4510984/108552693-9d50b700-732c-11eb-9ea6-c055f1c6ce8e.png)

**Comment by Holt59** [comment]
Then you probably need to upload your logs.

## Link to Mod Organizer logs:
Right click on the log window at the bottom of MO2, set log level to debug, restart Mo2 and reproduce the problem.
To access the log files you can press the Open folder icon over the modlist and select "Open MO2 Logs folder".

### MO Interface:
Create a [GIST](https://gist.github.com) which is a paste of your mo_interface logs, and link them here.
Do _NOT_ paste your full mo_interface logs here, as it will make this issue very long and hard to read!
If you are reporting a bug, _always_ include the mo_interface logs!

**Comment by GF-Huang** [comment]
https://gist.github.com/GF-Huang/c76f7720a767fec0818d756c68f85484

**Comment by Holt59** [comment]
The OMOD installer is loaded correctly in this log.

**Comment by GF-Huang** [comment]
I uninstall the re-install and click `load this plugin`, the issue disappeared.

**Comment by rulatir** [comment]
I am running with Lutris and I can't even get MO2 to run at all due to these bugs. There are 2 plugins that fail, "installer_omod.dll" and "installer_fomod_csharp.dll", and I click "Blacklist this plugin" but the blacklisting decision doesn't seem to be persisted because I keep getting these messages every time I start.

I installed MO2 using https://github.com/rockerbacon/lutris-skyrimse-installers , and it worked until I got clever and chose to allow it to self-update. It failed to restart after the update, and I am getting these messages ever since.

**Comment by isanae** [comment]
@rulatir What version of MO are you on?

**Comment by rulatir** [comment]
Was on 2.2.2.1, currently unsure because I can't get to the point where it would show me the current version, but It should be the newest that is available through online self-update because I just did that?

**Comment by isanae** [comment]
Can you come by the discord when you have a minute? Assuming you're on 2.4.0, there are no known issues with the blacklist. In the meantime, you can always downgrade to 2.2.2.1 if you want.

**Comment by rulatir** [comment]
I reinstalled 2.2.2.1 and it seems to be working.

**Comment by AnyOldName3** [comment]
Isa, bear in mind Lutris is mostly a Wine wrapper, so this is Linux. Here be dragons. As for the failure, my initial guess is Wine's got dodgy .NET FX 4.8 support or is accidentally configured such that it's just totally missing. Looking at the script in the linked repo, it seems only to be requesting the .NET Framework 4.0 from Winetricks, which is definitely below our minimum requirement.

**Comment by Cynthius** [comment]
I'm getting this issue too. .NET 4.8 is installed, and I've tried uninstalling and reinstalling MO2. I've also tried uninstalling and downgrading to 2.2.2.1 like rulatir mentioned. It no longer throws the error, but omods still don't show up in the downloads tab to install. They're immediately removed from the list once their download has finished.

Here's the GIST from the error: https://gist.github.com/Cynthius/f3fd34f039605b4f8bb35a4298d079f9

**Comment by AnyOldName3** [comment]
That's not actually got the error in it - it just says `user skipped plugin 'installer_omod.dll', remembering in loadcheck`, so presumably at some point in the past when you had the error, you clicked the button to not try loading it in the future. Can you go into MO2's settings, then to the plugins tab, and check the OMOD installer isn't in the blacklist or otherwise disabled, then restart MO2 and send us a new log, please?

**Comment by Cynthius** [comment]
Ah, sorry. Got a bit confused with the order I had to do things in. I've replaced it with a log that does have the error now. Not sure if that's bad practice or not.

Here's the link, for convenience: https://gist.github.com/Cynthius/f3fd34f039605b4f8bb35a4298d079f9

**Comment by AnyOldName3** [comment]
There's nothing particularly helpful in that log, either. It says a core dump was created, so maybe that'll tell us more.

**Comment by Cynthius** [comment]
How should I send the in core dump?

**Comment by AnyOldName3** [comment]
I guess zip it and attach it here.

**Comment by Cynthius** [comment]
Alright, here's the dump:

[Core Dump.zip](https://github.com/ModOrganizer2/modorganizer/files/6177636/Core.Dump.zip)

**Comment by AnyOldName3** [comment]
The main thread, which is the most likely to have died, doesn't seem to have anything on its stack bar one NT DLL function call, so I don't think that's going to be helpful, unfortunately.

**Comment by Cynthius** [comment]
Anything else I can do to help?

**Comment by mrudat** [comment]
Not certain if this is the same underlying error, but I'm also getting the same message against MO2 2.4.1, which I did not get against MO 2.4.0.

Here are two relevant events out of the event log that might help shed some light on what's going on:

Application error:
```
Faulting application name: ModOrganizer.exe, version: 2.4.1.0, time stamp: 0x6056203e
Faulting module name: KERNELBASE.dll, version: 10.0.19041.804, time stamp: 0x0e9c5eae
Exception code: 0xe0434352
Fault offset: 0x000000000002d759
Faulting process ID: 0x6210
Faulting application start time: 0x01d72376973778ce
Faulting application path: C:\Modding\MO2\ModOrganizer.exe
Faulting module path: C:\WINDOWS\System32\KERNELBASE.dll
Report ID: 58ed8909-5252-41e2-ad32-f865e24695b4
Faulting package full name: 
Faulting package-relative application ID: 
```

.Net Runtime:
```
Application: ModOrganizer.exe
Framework Version: v4.0.30319
Description: The process was terminated due to an unhandled exception.
Exception Info: System.NotSupportedException

Exception Info: System.IO.FileLoadException
   at System.Reflection.RuntimeAssembly._nLoad(System.Reflection.AssemblyName, System.String, System.Security.Policy.Evidence, System.Reflection.RuntimeAssembly, System.Threading.StackCrawlMark ByRef, IntPtr, Boolean, Boolean, Boolean)
   at System.Reflection.RuntimeAssembly.InternalLoadAssemblyName(System.Reflection.AssemblyName, System.Security.Policy.Evidence, System.Reflection.RuntimeAssembly, System.Threading.StackCrawlMark ByRef, IntPtr, Boolean, Boolean, Boolean)
   at System.Reflection.RuntimeAssembly.InternalLoadFrom(System.String, System.Security.Policy.Evidence, Byte[], System.Configuration.Assemblies.AssemblyHashAlgorithm, Boolean, Boolean, System.Threading.StackCrawlMark ByRef)
   at System.Reflection.Assembly.LoadFrom(System.String)
   at <Module>.AssemblyResolver.OnAssemblyResolve(System.Object, System.ResolveEventArgs)
   at System.AppDomain.OnAssemblyResolveEvent(System.Reflection.RuntimeAssembly, System.String)

Exception Info: System.IO.FileLoadException
   at <Module>.OMODFrameworkWrapper.constructorHelper(OMODFrameworkWrapper*)
   at <Module>.OMODFrameworkWrapper.{ctor}(OMODFrameworkWrapper*, MOBase.IOrganizer*, QWidget*)
   at <Module>.std.make_unique<class OMODFrameworkWrapper,class MOBase::IOrganizer * &,0>(std.unique_ptr<OMODFrameworkWrapper,std::default_delete<OMODFrameworkWrapper> >*, MOBase.IOrganizer**)
   at <Module>.InstallerOMOD.init(InstallerOMOD*, MOBase.IOrganizer*)
```

**Comment by AnyOldName3** [comment]
What do you get if you run this in PowerShell?
```powershell
Get-ChildItem HKLM:\SOFTWARE\Microsoft\.NETFramework\v4.0.30319\SKUs\
```

**Comment by mrudat** [comment]
> What do you get if you run this in PowerShell?
> 
> ```powershell
> Get-ChildItem HKLM:\SOFTWARE\Microsoft\.NETFramework\v4.0.30319\SKUs\
> ```
I've got VS 2019 installed, so I have many versions of .NETFramework installed, both SDK and Client.
```
> ls HKLM:\SOFTWARE\Microsoft\.NETFramework\v4.0.30319\SKUs\ | FT Name
HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\.NETFramework\v4.0.30319\SKUs\.NETFramework,Version=v4.0
HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\.NETFramework\v4.0.30319\SKUs\.NETFramework,Version=v4.0,Profile=Client
HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\.NETFramework\v4.0.30319\SKUs\.NETFramework,Version=v4.0.1
HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\.NETFramework\v4.0.30319\SKUs\.NETFramework,Version=v4.0.1,Profile=Client
HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\.NETFramework\v4.0.30319\SKUs\.NETFramework,Version=v4.0.2
HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\.NETFramework\v4.0.30319\SKUs\.NETFramework,Version=v4.0.2,Profile=Client
HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\.NETFramework\v4.0.30319\SKUs\.NETFramework,Version=v4.0.3
HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\.NETFramework\v4.0.30319\SKUs\.NETFramework,Version=v4.0.3,Profile=Client
HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\.NETFramework\v4.0.30319\SKUs\.NETFramework,Version=v4.5
HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\.NETFramework\v4.0.30319\SKUs\.NETFramework,Version=v4.5.1
HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\.NETFramework\v4.0.30319\SKUs\.NETFramework,Version=v4.5.2
HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\.NETFramework\v4.0.30319\SKUs\.NETFramework,Version=v4.5.3
HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\.NETFramework\v4.0.30319\SKUs\.NETFramework,Version=v4.6
HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\.NETFramework\v4.0.30319\SKUs\.NETFramework,Version=v4.6.1
HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\.NETFramework\v4.0.30319\SKUs\.NETFramework,Version=v4.6.2
HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\.NETFramework\v4.0.30319\SKUs\.NETFramework,Version=v4.7
HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\.NETFramework\v4.0.30319\SKUs\.NETFramework,Version=v4.7.1
HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\.NETFramework\v4.0.30319\SKUs\.NETFramework,Version=v4.7.2
HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\.NETFramework\v4.0.30319\SKUs\.NETFramework,Version=v4.8
HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\.NETFramework\v4.0.30319\SKUs\Client
HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\.NETFramework\v4.0.30319\SKUs\Default
```

**Comment by AnyOldName3** [comment]
So literally the same as I've got, and it works for me.

If you've got VS installed, I guess you could try debugging MO2. We've got PDBs somewhere. Hopefully, with the mode set to mixed (for some reason, with auto, it doesn't notice the .NET stuff) you can get it to break on the right exception and might see something helpful, e.g. it having loaded a much-too-old .NET Framework DLL, despite you having 4.8 installed.

**Comment by mrudat** [comment]
After reinstalling my graphics card drivers, steam, and rebooting my computer a silly number of times... it looks like it's working (for now).

**Comment by josephsmendoza** [comment]
I'm still having this issue on 2.4.2, windows 10 19041. Same unhelpful log with core dump.

**Comment by AnyOldName3** [comment]
Depending on what caused it to die, there's a change that didn't quite make it into 2.4.2 that should make it log a few more potential errors instead of totally exploding.

**Comment by josephsmendoza** [comment]
Got a pr/commit to reference for that?

**Comment by LostDragonist** [comment]
Probably https://github.com/ModOrganizer2/modorganizer-installer_omod/pull/3 or https://github.com/ModOrganizer2/modorganizer-installer_omod/pull/4

**Comment by IsaacSin** [comment]
I was encountering this problem, then I remembered that I had changed the install directory to be under Program Files. I ran ModOrganizer.exe as admin and it loaded properly. I then promptly reinstalled under the default directory and it worked without admin. 

Looking at Cynthius's log, they also had it installed under Program Files, so it might have been the same case. mrudat had it under the default, though, so that's not all that's going on here.

**Comment by josephsmendoza** [comment]
I also keep mod organizer in program files. Sounds like it might be UAC related.

**Comment by AnyOldName3** [comment]
In general, if software doesn't work, it's a really bad idea to run it as Admin. It's demonstrated it's not capable of doing what you tell it, and suddenly you're giving it more power. It's analogous to giving a child a submachine gun because you were annoyed that they weren't picking up their Nerf darts. Unfortunately, Microsoft accidentally trained people to include it in their troubleshooting when they rushed Vista out the door without giving developers enough time to adapt their applications to work with it.

**Comment by josephsmendoza** [comment]
It's just a troubleshooting step, we're not keeping it that way. It seems the solution is probably to follow the advice to install MO outside of program files. Issue closed?

**Comment by AnyOldName3** [comment]
Nah, something's broken and now we have repro steps, so hopefully it'll be a simple fix. Having MO2 out of Program Files will work as a workaround until another release happens, though.

**Comment by AnyOldName3** [comment]
```
System.TypeInitializationException
  HResult=0x80131534
  Message=The type initializer for 'OMODFramework.Utils' threw an exception.
  Source=installer_omod
  StackTrace:
   at OMODFrameworkWrapper.initFrameworkSettings(OMODFrameworkWrapper* )
   at OMODFrameworkWrapper.{ctor}(OMODFrameworkWrapper* , IOrganizer* organizer, QWidget* parentWidget)
   at std.make_unique<class OMODFrameworkWrapper,class MOBase::IOrganizer * &,0>(unique_ptr<OMODFrameworkWrapper\,std::default_delete<OMODFrameworkWrapper> >* , IOrganizer** <_Args_0>)
   at InstallerOMOD.init(InstallerOMOD* , IOrganizer* moInfo)

  This exception was originally thrown at this call stack:
    OMODFramework.FileLogger.Init()
    OMODFramework.Utils.Utils()

Inner Exception 1:
OMODFrameworkException: Could not create logging file at C:\Program Files\MO2\plugins\data\OMODFramework.log
System.UnauthorizedAccessException: Access to the path 'C:\Program Files\MO2\plugins\data\OMODFramework.log' is denied.
   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize)
   at OMODFramework.FileLogger.Init()
```

**Comment by AnyOldName3** [comment]
https://github.com/ModOrganizer2/modorganizer-installer_omod/pull/5 should fix this.

**Comment by PopeRigby** [comment]
I'm having this issue on Linux, but MO2 is installed in `C:\Modding\MO2` in my Wine prefix, and not Program Files. Is it the same problem?

**Comment by AnyOldName3** [comment]
No. There's an issue where Wine doesn't know we need the .NET framework 4.8, but I can't remember whether that's the only issue.

**Comment by PopeRigby** [comment]
Ah. It might have something to do with .NET 4.8 being broken since WINE 6.10

```
This package (dotnet48) is broken in wine-6.10. Broken since 5.18. See https://bugs.winehq.org/show_bug.cgi?id=49897 for more info. Use --force to try anyway.
```

**Comment by PopeRigby** [comment]
@AnyOldName3 so, I was able to install .NET 4.8, but it causes WINE to crash:
```
0108:err:kerberos:kerberos_LsaApInitializePackage no Kerberos support, expect problems
0108:err:winediag:check_version ntlm_auth was not found or is outdated. Make sure that ntlm_auth >= 3.0.25 is in your path. Usually, you can find it in the winbind package of your distribution.
0108:err:ntlm:ntlm_LsaApInitializePackage no NTLM support, expect problems
[2021-07-02 03:08:36.680 D] timing: MOApplication() 0 ms
[2021-07-02 03:08:36.681 D] timing: main() multiprocess 112 ms
[2021-07-02 03:08:36.685 D] looking for plugins in C:\Modding\MO2\plugins
[2021-07-02 03:08:36.686 D] loaded plugin 'BSA Extractor' from 'bsa_extractor.dll' - [Plugin]
[2021-07-02 03:08:36.693 D] loaded plugin 'BSA Packer' from 'bsa_packer.dll' - [Tool]
[2021-07-02 03:08:36.694 D] loaded plugin 'FNIS Checker' from 'check_fnis.dll' - [Plugin]
[2021-07-02 03:08:36.695 D] loaded plugin 'Basic diagnosis plugin' from 'diagnose_basic.dll' - [Diagnose]
[2021-07-02 03:08:36.697 D] loaded plugin 'Enderal Support Plugin' from 'game_enderal.dll' - [Game, File Mapper]
[2021-07-02 03:08:36.698 D] loaded plugin 'Enderal Special Edition Support Plugin' from 'game_enderalse.dll' - [Game, File Mapper]
[2021-07-02 03:08:36.700 D] loaded plugin 'Fallout 3 Support Plugin' from 'game_fallout3.dll' - [Game, File Mapper]
[2021-07-02 03:08:36.702 D] loaded plugin 'Fallout 4 Support Plugin' from 'game_fallout4.dll' - [Game, File Mapper]
[2021-07-02 03:08:36.704 D] loaded plugin 'Fallout 4 VR Support Plugin' from 'game_fallout4vr.dll' - [Game, File Mapper]
[2021-07-02 03:08:36.706 D] loaded plugin 'Fallout NV Support Plugin' from 'game_falloutNV.dll' - [Game, File Mapper]
[2021-07-02 03:08:36.707 D] loaded plugin 'Morrowind Support Plugin' from 'game_morrowind.dll' - [Game, File Mapper]
[2021-07-02 03:08:36.709 D] loaded plugin 'Oblivion Support Plugin' from 'game_oblivion.dll' - [Game, File Mapper]
[2021-07-02 03:08:36.710 D] loaded plugin 'Skyrim Support Plugin' from 'game_skyrim.dll' - [Game, File Mapper]
[2021-07-02 03:08:36.712 D] loaded plugin 'Skyrim Special Edition Support Plugin' from 'game_skyrimse.dll' - [Game, File Mapper]
[2021-07-02 03:08:36.714 D] loaded plugin 'Skyrim VR Support Plugin' from 'game_skyrimvr.dll' - [Game, File Mapper]
[2021-07-02 03:08:36.716 D] loaded plugin 'Fallout TTW Support Plugin' from 'game_ttw.dll' - [Game, File Mapper]
[2021-07-02 03:08:36.717 D] loaded plugin 'INI Bakery' from 'inibakery.dll' - [File Mapper]
[2021-07-02 03:08:36.718 D] loaded plugin 'INI Editor' from 'inieditor.dll' - [Tool]
[2021-07-02 03:08:36.719 D] loaded plugin 'BAIN Installer' from 'installer_bain.dll' - [Installer]
[2021-07-02 03:08:36.720 D] loaded plugin 'Bundle Installer' from 'installer_bundle.dll' - [Installer]
[2021-07-02 03:08:36.721 D] loaded plugin 'Fomod Installer' from 'installer_fomod.dll' - [Installer, Diagnose]
0108:err:mscoree:LoadLibraryShim error reading registry key for installroot
0108:err:mscoree:CLRRuntimeInfo_GetRuntimeDirectory error reading registry key for installroot
0108:err:mscoree:ReallyFixupVTable unable to fixup vtable, hr=80004005, status=0
0108:err:mscoree:expect_no_runtimes Process exited with a Mono runtime loaded.
```

**Comment by AnyOldName3** [comment]
That's definitely a WINE problem rather than a me problem. Presumably, they're already tracking the lack of 4.8 support so there's not much we can do to speed it up.

**Comment by PopeRigby** [comment]
I don't actually, see [any open bugs about 4.8](https://bugs.winehq.org/buglist.cgi?quicksearch=dotnet48), and the one I linked earlier is closed,

**Comment by AnyOldName3** [comment]
I just googled the last line of the error and a vaguely similar issue report for different software said that the only supported way to install the .NET framework for WINE is via WineTricks. Is that how you installed it, or did you just run the Windows installer under WINE?

**Comment by PopeRigby** [comment]
I just ran the Windows installer under WINE. Can you send me that link, please?

**Comment by AnyOldName3** [comment]
https://forum.winehq.org/viewtopic.php?t=32302#p122004

**Comment by PopeRigby** [comment]
Maybe I should file a WINE bug report.

**Comment by eddwarddd** [comment]
Hey, I fixed this by installing dotnet 4.8 through  `winetricks -q dotnet48` , runs perfectly now.

**Comment by PopeRigby** [comment]
@eddwarddd Did you have to force it? It gives me this error:
```
warning: This package (dotnet48) is broken in wine-6.12. Broken since 5.18. See https://bugs.winehq.org/show_bug.cgi?id=49897 for more info. Use --force to try anyway.
```

**Comment by PopeRigby** [comment]
Nevermind. Forcing it, did in fact work.

**Comment by Kelniimir** [comment]
Passerby here.

Updated 2.4.0->2.4.2 (rev. 2a5dd6c7).
NetFrameworks installed, ndp48 could install because "newer one" already installed.
MO was installed in C:\ProgramFiles\MO2, therefore update was installed there as well.
Issue occurred.
Downloading MO 2.4.2 installed from Nexus and overwriting whilst "as admin" did not helped.
Uninstalling, installing on "new" default path C:\Modding\MO2 helped (although at fist launch, plugin was still making error).
![image](https://user-images.githubusercontent.com/34988975/127775338-bf714403-7093-4cb3-8397-87dd4e5d5426.png)

It seems it worked for me...although option to enable it is greyed out.

**Comment by Cynthius** [comment]
> 
> 
> I experienced same error on Windows 10 Mod Organizer version 2.4.2.0 if
> 
>     1. Start Mod Organizer with admin privileges
> 
>     2. Start Mod Organizer without admin privileges
> 
>     3. Error occurs
> 
>     4. Starting Mod Organizer with admin privileges again works fine
> 
> 
> Suspecting file access privileges changes after starting with admin privileges causing the issue.

Can confirm that launching in Admin mode resolves the issue for me too, but still throws the error any time I run without admin privileges.

**Comment by AnyOldName3** [comment]
Don't start broken software in admin mode. That's like giving a toddler who's shot you with a nerf gun an assault rifle.

As for the requires-admin-when-MO2-is-installed-to-program-files issue, it's resolved in the 2.4.3 alpha builds available from our development Discord server.

**Comment by Bananaman** [comment]
This is the only correct person here:

> Changing 'Users' permissions to include modify and write access to MO2\ModOrganizer.ini fixed the issue, MO2 starts as expected without admin privileges ![image](https://user-images.githubusercontent.com/881361/136821647-3dc5b6a5-57c2-4ab2-80d0-a552fc7d9e9d.png)

Here's what to do:

- Go to the folder where you installed Mod Organizer.
- Right click that folder, and go edit the security/permissions.
- Give "Users" (regular users) full permissions, meaning write access too.

That's it. That is the issue. Mod Organizer installs itself without giving itself permission to write to its own folders. It's an easy fix which @isanae @Holt59 @Al12rs @LostDragonist can probably be integrated into the installer automatically.

**Comment by AnyOldName3** [comment]
Putting a portable instance in program files is absolutely not supported. There are a bazillion potential permissions issues that could cause, and working around them can introduce security holes, so we're definitely not going to meddle with permissions in Program Files.

With a normally-configured Windows machine, users should already have full control of arbitrary loose directories in the root of the boot drive (provided those directories don't have explicit protection like Program Files does), so there should be no need for the installer to explicitly set that up. Do you perhaps have a non-default setting that changes this? Is it a particularly recent Windows installation (in which case Microsoft might have changed the rules).

**Comment by MrStonedOne** [comment]
I can confirm this issue occurs on a machine with UAC disabled and full permissions while running under an administrator account.

Its not that.

**Comment by AnyOldName3** [comment]
> I can confirm this issue occurs on a windows machine with UAC disabled and full permissions while running under an administrator account.
> 
> Its not that.

Then that's a different issue with similar symptoms, so should be tracked separately. Also, scroll up a bit so I don't need to type out the toddler-with-an-uzi analogy again, because it's already like half the posts here.

Can you open a separate issue report and include all the logs and things it suggests, please? It might also be a good idea to try both the latest release and the current beta/dev build, as I think there's some extra logging in the newer build.

**Comment by Cenness** [comment]
For a blocked file variant on windows:

```
File name: 'file:///[REDACTED]\mo2\plugins\data\OMODFramework.dll' --->
System.NotSupportedException: An attempt was made to load an assembly from a network location which
would have caused the assembly to be sandboxed in previous versions of the .NET Framework.
This release of the .NET Framework does not enable CAS policy by default, so this load may be dangerous.
If this load is not intended to sandbox the assembly, please enable the loadFromRemoteSources switch.
See http://go.microsoft.com/fwlink/?LinkId=155569 for more information.
```

Solution is following:
1. start powershell
2. cd to mo2 installation folder
3. run `dir -Recurse | Unblock-File`

**Comment by Jownoo** [comment]
Not sure if this has already been said, but I had the same issue. Try and completely uninstall Mod Organizer, and all of the files other than the original 4 7z files for the game. Then try again but make sure you unzip the files with 7zip, not Winrar or Winzip. Hope this helps.

**Comment by 8day** [comment]
Had the same issue with 2.4.4 *and* 2.4.0. Couldn't fix by unpacking archive anew as well as replacing existing data.

I think I might've fixed it either by doing one of these, or all of them:

- did proper installation through executable;
- installed with admin rights;
- disabled HIPS in my security suite (set it to training mode). Something similar happened before due to antivirus, so I think this one may have caused the problem.

**Comment by AnyOldName3** [comment]
It's not a new addition. It's been there the whole time we've had an OMOD installer. It *has* been updated a couple of times, so it's possible Windows Defender has taken a dislike to a new version, but it should be more trustworthy than any of the other DLLs that come with MO2 as it comes from Nuget, which Microsoft manage, and if they didn't like it, they'd surely remove it instead of distributing it to people.

**Comment by jaspervanbrian** [comment]
This bug is still happening for me when trying to install the Portable Version (v2.5+)
I am on Windows 11, and I am trying to install it for New Vegas (Epic Games variant).

**Comment by CritLoren** [comment]
I think the only way they can fix this is make it a self extracting archive or make the installer have a portable install option.

Before you unarchive the portable zip right click it, go to properties, and click unblock. Now you can extract mo2 and it should work fine now.

**Comment by Holt59** [comment]
> I think the only way they can fix this is make it a self extracting archive or make the installer have a portable install option.
> 
> Before you unarchive the portable zip right click it, go to properties, and click unblock. Now you can extract mo2 and it should work fine now.

The installer already has a portable option. The archive has been deprecated for a long time, it's here for other tools such as Wabbajack but there is no reason to use it to install MO2 as a standard user.

**Comment by taylorjdlee** [comment]
> For a blocked file variant on windows:
> 
> ```
> File name: 'file:///[REDACTED]\mo2\plugins\data\OMODFramework.dll' --->
> System.NotSupportedException: An attempt was made to load an assembly from a network location which
> would have caused the assembly to be sandboxed in previous versions of the .NET Framework.
> This release of the .NET Framework does not enable CAS policy by default, so this load may be dangerous.
> If this load is not intended to sandbox the assembly, please enable the loadFromRemoteSources switch.
> See http://go.microsoft.com/fwlink/?LinkId=155569 for more information.
> ```
> 
> Solution is following:
> 
> 1. start powershell
> 2. cd to mo2 installation folder
> 3. run `dir -Recurse | Unblock-File`

Beautiful, this just worked for me! Thanks for your help!

**Comment by AJScanlan** [comment]
If you're following the Viva New Vegas guide do not skip step 3 when installing MO2 as it will cause this issue

Step 3 specifically is Unblocking the 7z file in the file properties

https://vivanewvegas.moddinglinked.com/mo2.html

**Comment by hughesjr** [comment]
> > For a blocked file variant on windows:
> > ```
> > File name: 'file:///[REDACTED]\mo2\plugins\data\OMODFramework.dll' --->
> > System.NotSupportedException: An attempt was made to load an assembly from a network location which
> > would have caused the assembly to be sandboxed in previous versions of the .NET Framework.
> > This release of the .NET Framework does not enable CAS policy by default, so this load may be dangerous.
> > If this load is not intended to sandbox the assembly, please enable the loadFromRemoteSources switch.
> > See http://go.microsoft.com/fwlink/?LinkId=155569 for more information.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Solution is following:
> > 
> > 1. start powershell
> > 2. cd to mo2 installation folder
> > 3. run `dir -Recurse | Unblock-File`
> 
> Beautiful, this just worked for me! Thanks for your help!

This also worked on an install I just did to resolve the same error.

**Comment by Jake2614** [comment]
Hey so I'm mew to PC stuff but i keep getting this error and I've tried some of the solutions above but they didn't work.![Image](https://github.com/user-attachments/assets/227b8063-b432-4083-b572-ab81904b9ec7)

**Comment by AnyOldName3** [comment]
Did you install MO2 with its installer like the instructions above said to, or with the archive, like the instructions above said not to? That error message shouldn't be possible if you used the installer.

**Comment by Jake2614** [comment]
> Did you install MO2 with its installer like the instructions above said to, or with the archive, like the instructions above said not to? That error message shouldn't be possible if you used the installer.

I figured out the problem. I had failed to actually install and setup 7zip and instead extracted MO2 using WinRAR (rookie mistake) so i uninstalled 7Zip and MO2 and correctly reset up everything and it worked. It was completely user error but I am also very new to all of this.

**Comment by AnyOldName3** [comment]
That still sounds like you're using the archive instead of the installer (the installer's self-extracting, so doesn't need either of those tools), which is asking for trouble for more reasons than just this.

**Comment by Tamagavka** [comment]
![Image](https://github.com/user-attachments/assets/e01a4da2-e3cd-4ef6-88b1-9fd8496b5488) Today I encountered the same problem, on the game cyberpunk, a new framework is installed, I launched it from the installer, other people have this build working, judging by the comments, installed it in the game directory, and not on the system disk

**Comment by Kiaos** [comment]
`[2025-03-12 03:35:22.251 D] command line: '"D:\Modding\Sin1ster Valheim (Mo2)\ModOrganizer.exe" '
[2025-03-12 03:35:22.251 I] starting Mod Organizer version 2.5.2 revision 9c130cbf in D:/Modding/Sin1ster Valheim (Mo2), usvfs: 0.5.6.1
[2025-03-12 03:35:22.251 I] data path: D:/Modding/Sin1ster Valheim (Mo2)
[2025-03-12 03:35:22.251 I] working directory: D:/Modding/Sin1ster Valheim (Mo2)
[2025-03-12 03:35:22.251 D] timing: MOApplication setup() 23875 ms
[2025-03-12 03:35:22.447 W] 'D:/Modding/Sin1ster Valheim (Mo2)/helper.exe': file is blocked (Internet (3))
[2025-03-12 03:35:22.447 W] 'D:/Modding/Sin1ster Valheim (Mo2)/libcrypto-3-x64.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.448 W] 'D:/Modding/Sin1ster Valheim (Mo2)/libssl-3-x64.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.448 W] 'D:/Modding/Sin1ster Valheim (Mo2)/QtWebEngineProcess.exe': file is blocked (Internet (3))
[2025-03-12 03:35:22.448 W] 'D:/Modding/Sin1ster Valheim (Mo2)/uibase.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.448 W] 'D:/Modding/Sin1ster Valheim (Mo2)/usvfs_proxy_x64.exe': file is blocked (Internet (3))
[2025-03-12 03:35:22.448 W] 'D:/Modding/Sin1ster Valheim (Mo2)/usvfs_proxy_x86.exe': file is blocked (Internet (3))
[2025-03-12 03:35:22.449 W] 'D:/Modding/Sin1ster Valheim (Mo2)/usvfs_x64.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.449 W] 'D:/Modding/Sin1ster Valheim (Mo2)/usvfs_x86.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.449 W] 'D:/Modding/Sin1ster Valheim (Mo2)/dlls/7z.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.449 W] 'D:/Modding/Sin1ster Valheim (Mo2)/dlls/archive.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.450 W] 'D:/Modding/Sin1ster Valheim (Mo2)/dlls/d3dcompiler_47.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.450 W] 'D:/Modding/Sin1ster Valheim (Mo2)/dlls/dxcompiler.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.450 W] 'D:/Modding/Sin1ster Valheim (Mo2)/dlls/dxil.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.450 W] 'D:/Modding/Sin1ster Valheim (Mo2)/dlls/libbsarch.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.450 W] 'D:/Modding/Sin1ster Valheim (Mo2)/dlls/libcrypto-3-x64.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.451 W] 'D:/Modding/Sin1ster Valheim (Mo2)/dlls/liblz4.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.451 W] 'D:/Modding/Sin1ster Valheim (Mo2)/dlls/libssl-3-x64.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.451 W] 'D:/Modding/Sin1ster Valheim (Mo2)/dlls/opengl32sw.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.451 W] 'D:/Modding/Sin1ster Valheim (Mo2)/dlls/Qt6Concurrent.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.451 W] 'D:/Modding/Sin1ster Valheim (Mo2)/dlls/Qt6Core.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.452 W] 'D:/Modding/Sin1ster Valheim (Mo2)/dlls/Qt6Gui.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.452 W] 'D:/Modding/Sin1ster Valheim (Mo2)/dlls/Qt6Network.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.452 W] 'D:/Modding/Sin1ster Valheim (Mo2)/dlls/Qt6OpenGL.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.452 W] 'D:/Modding/Sin1ster Valheim (Mo2)/dlls/Qt6OpenGLWidgets.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.452 W] 'D:/Modding/Sin1ster Valheim (Mo2)/dlls/Qt6Positioning.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.452 W] 'D:/Modding/Sin1ster Valheim (Mo2)/dlls/Qt6PrintSupport.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.453 W] 'D:/Modding/Sin1ster Valheim (Mo2)/dlls/Qt6Qml.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.453 W] 'D:/Modding/Sin1ster Valheim (Mo2)/dlls/Qt6QmlModels.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.453 W] 'D:/Modding/Sin1ster Valheim (Mo2)/dlls/Qt6QmlWorkerScript.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.453 W] 'D:/Modding/Sin1ster Valheim (Mo2)/dlls/Qt6Quick.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.453 W] 'D:/Modding/Sin1ster Valheim (Mo2)/dlls/Qt6Quick3DUtils.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.454 W] 'D:/Modding/Sin1ster Valheim (Mo2)/dlls/Qt6QuickWidgets.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.454 W] 'D:/Modding/Sin1ster Valheim (Mo2)/dlls/Qt6SerialPort.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.454 W] 'D:/Modding/Sin1ster Valheim (Mo2)/dlls/Qt6Svg.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.454 W] 'D:/Modding/Sin1ster Valheim (Mo2)/dlls/Qt6WebChannel.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.454 W] 'D:/Modding/Sin1ster Valheim (Mo2)/dlls/Qt6WebEngineCore.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.455 W] 'D:/Modding/Sin1ster Valheim (Mo2)/dlls/Qt6WebEngineWidgets.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.455 W] 'D:/Modding/Sin1ster Valheim (Mo2)/dlls/Qt6WebSockets.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.455 W] 'D:/Modding/Sin1ster Valheim (Mo2)/dlls/Qt6Widgets.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.455 W] 'D:/Modding/Sin1ster Valheim (Mo2)/loot/loot.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.456 W] 'D:/Modding/Sin1ster Valheim (Mo2)/loot/lootcli.exe': file is blocked (Internet (3))
[2025-03-12 03:35:22.456 W] 'D:/Modding/Sin1ster Valheim (Mo2)/helper.exe': file is blocked (Internet (3))
[2025-03-12 03:35:22.456 W] 'D:/Modding/Sin1ster Valheim (Mo2)/libcrypto-3-x64.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.456 W] 'D:/Modding/Sin1ster Valheim (Mo2)/libssl-3-x64.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.457 W] 'D:/Modding/Sin1ster Valheim (Mo2)/QtWebEngineProcess.exe': file is blocked (Internet (3))
[2025-03-12 03:35:22.457 W] 'D:/Modding/Sin1ster Valheim (Mo2)/uibase.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.457 W] 'D:/Modding/Sin1ster Valheim (Mo2)/usvfs_proxy_x64.exe': file is blocked (Internet (3))
[2025-03-12 03:35:22.457 W] 'D:/Modding/Sin1ster Valheim (Mo2)/usvfs_proxy_x86.exe': file is blocked (Internet (3))
[2025-03-12 03:35:22.457 W] 'D:/Modding/Sin1ster Valheim (Mo2)/usvfs_x64.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.457 W] 'D:/Modding/Sin1ster Valheim (Mo2)/usvfs_x86.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.458 W] 'D:/Modding/Sin1ster Valheim (Mo2)/platforms/qwindows.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.458 W] 'D:/Modding/Sin1ster Valheim (Mo2)/plugins/bsa_extractor.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.458 W] 'D:/Modding/Sin1ster Valheim (Mo2)/plugins/bsa_packer.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.459 W] 'D:/Modding/Sin1ster Valheim (Mo2)/plugins/check_fnis.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.459 W] 'D:/Modding/Sin1ster Valheim (Mo2)/plugins/diagnose_basic.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.459 W] 'D:/Modding/Sin1ster Valheim (Mo2)/plugins/game_enderal.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.459 W] 'D:/Modding/Sin1ster Valheim (Mo2)/plugins/game_enderalse.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.459 W] 'D:/Modding/Sin1ster Valheim (Mo2)/plugins/game_fallout3.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.460 W] 'D:/Modding/Sin1ster Valheim (Mo2)/plugins/game_fallout4.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.460 W] 'D:/Modding/Sin1ster Valheim (Mo2)/plugins/game_fallout4vr.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.460 W] 'D:/Modding/Sin1ster Valheim (Mo2)/plugins/game_fallout76.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.460 W] 'D:/Modding/Sin1ster Valheim (Mo2)/plugins/game_falloutNV.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.460 W] 'D:/Modding/Sin1ster Valheim (Mo2)/plugins/game_morrowind.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.461 W] 'D:/Modding/Sin1ster Valheim (Mo2)/plugins/game_nehrim.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.461 W] 'D:/Modding/Sin1ster Valheim (Mo2)/plugins/game_oblivion.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.461 W] 'D:/Modding/Sin1ster Valheim (Mo2)/plugins/game_skyrim.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.461 W] 'D:/Modding/Sin1ster Valheim (Mo2)/plugins/game_skyrimse.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.461 W] 'D:/Modding/Sin1ster Valheim (Mo2)/plugins/game_skyrimvr.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.462 W] 'D:/Modding/Sin1ster Valheim (Mo2)/plugins/game_starfield.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.462 W] 'D:/Modding/Sin1ster Valheim (Mo2)/plugins/game_ttw.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.462 W] 'D:/Modding/Sin1ster Valheim (Mo2)/plugins/inibakery.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.462 W] 'D:/Modding/Sin1ster Valheim (Mo2)/plugins/inieditor.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.462 W] 'D:/Modding/Sin1ster Valheim (Mo2)/plugins/installer_bain.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.463 W] 'D:/Modding/Sin1ster Valheim (Mo2)/plugins/installer_bundle.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.463 W] 'D:/Modding/Sin1ster Valheim (Mo2)/plugins/installer_fomod.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.463 W] 'D:/Modding/Sin1ster Valheim (Mo2)/plugins/installer_fomod_csharp.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.464 W] 'D:/Modding/Sin1ster Valheim (Mo2)/plugins/installer_manual.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.464 W] 'D:/Modding/Sin1ster Valheim (Mo2)/plugins/installer_quick.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.464 W] 'D:/Modding/Sin1ster Valheim (Mo2)/plugins/preview_base.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.464 W] 'D:/Modding/Sin1ster Valheim (Mo2)/plugins/preview_bsa.dll': file is blocked (Internet (3))
[2025-03-12 03:35:22.483 W] loadcheck file found for plugin 'installer_omod'
[2025-03-12 03:37:34.767 W] user wants to load plugin 'installer_omod' anyway
[2025-03-12 03:37:34.815 E] Unhanded .NET Exception: System.IO.FileLoadException: Could not load file or assembly 'OMODFramework, Version=2.2.2.0, Culture=neutral, PublicKeyToken=null' or one of its dependencies. Operation is not supported. (Exception from HRESULT: 0x80131515)

[2025-03-12 03:37:34.815 E] File name: 'OMODFramework, Version=2.2.2.0, Culture=neutral, PublicKeyToken=null' ---> System.IO.FileLoadException: Could not load file or assembly 'file:///D:\Modding\Sin1ster Valheim (Mo2)\plugins\installer_omod\OMODFramework.dll' or one of its dependencies. Operation is not supported. (Exception from HRESULT: 0x80131515)

[2025-03-12 03:37:34.815 E] File name: 'file:///D:\Modding\Sin1ster Valheim (Mo2)\plugins\installer_omod\OMODFramework.dll' ---> System.NotSupportedException: An attempt was made to load an assembly from a network location which would have caused the assembly to be sandboxed in previous versions of the .NET Framework. This release of the .NET Framework does not enable CAS policy by default, so this load may be dangerous. If this load is not intended to sandbox the assembly, please enable the loadFromRemoteSources switch. See http://go.microsoft.com/fwlink/?LinkId=155569 for more information.

[2025-03-12 03:37:34.815 E]    at System.Reflection.RuntimeAssembly._nLoad(AssemblyName fileName, String codeBase, Evidence assemblySecurity, RuntimeAssembly locationHint, StackCrawlMark& stackMark, IntPtr pPrivHostBinder, Boolean throwOnFileNotFound, Boolean forIntrospection, Boolean suppressSecurityChecks)

[2025-03-12 03:37:34.815 E]    at System.Reflection.RuntimeAssembly.InternalLoadAssemblyName(AssemblyName assemblyRef, Evidence assemblySecurity, RuntimeAssembly reqAssembly, StackCrawlMark& stackMark, IntPtr pPrivHostBinder, Boolean throwOnFileNotFound, Boolean forIntrospection, Boolean suppressSecurityChecks)

[2025-03-12 03:37:34.815 E]    at System.Reflection.RuntimeAssembly.InternalLoadFrom(String assemblyFile, Evidence securityEvidence, Byte[] hashValue, AssemblyHashAlgorithm hashAlgorithm, Boolean forIntrospection, Boolean suppressSecurityChecks, StackCrawlMark& stackMark)

[2025-03-12 03:37:34.815 E]    at System.Reflection.Assembly.LoadFrom(String assemblyFile)

[2025-03-12 03:37:34.815 E]    at AssemblyResolver.OnAssemblyResolve(Object sender, ResolveEventArgs args)

[2025-03-12 03:37:34.815 E]    at System.AppDomain.OnAssemblyResolveEvent(RuntimeAssembly assembly, String assemblyFullName)

[2025-03-12 03:37:34.815 E] 

[2025-03-12 03:37:34.815 E] 

[2025-03-12 03:37:34.815 E]    at OMODFrameworkWrapper.constructorHelper(OMODFrameworkWrapper* )

[2025-03-12 03:37:34.815 E]    at OMODFrameworkWrapper.{ctor}(OMODFrameworkWrapper* , IOrganizer* organizer, QWidget* parentWidget)

[2025-03-12 03:37:34.815 E] 

[2025-03-12 03:37:34.815 E] 
`
